### ISSUE TITLE
cannot include dtype 'M' in a buffer

### ISSUE THREAD

**Issue by mangecoeur** [joblib/joblib#183]
Using memory cache with pandas dataframes with datelike columns results in error `cannot include dtype 'M' in a buffer`

Suspect it is related to this issue:

https://github.com/pydata/pandas/issues/4533

**Comment by lesteve** [comment]
Duplicate of #188. Can someone close it?

Note that the problem happens both with dtype='datetime64' ('M8') and dtype='timedelta' ('m8').

A snippet to reproduce the problem and show that the problem happens when the input is hashed:

```python

import joblib

mem = joblib.Memory('/tmp/joblib')
import pandas as pd

def make_dataframe_with_timelike_columns():
    df = pd.util.testing.makeDataFrame()
    df.A = pd.to_datetime(df.A)
    return df

@mem.cache
def identity(df):
    return df

df = make_dataframe_with_timelike_columns()
identity(df)
```

**Comment by GaelVaroquaux** [comment]
I am closing this, but maybe you should include the snipping on #188.

**Comment by lesteve** [comment]
FWIW there is already a more minimalistic snippet available on #188.
